boot/freeldr/freeldr/arch/i386/finishRelocation.S

```diff
@@ -0,0 +1,65 @@
+/*
+ * PROJECT:     FreeLoader
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Finish the partial relocation to the excepted address.
+ * COPYRIGHT:   Copyright 2024 Daniel Victor <ilauncherdeveloper@gmail.com>
+ */
+
+#include <asm.inc>
+#include <arch/pc/x86common.h>
+
+#ifdef _M_IX86
+#define rsi esi
+#define rdi edi
+#define rcx ecx
+#define qword dword
+#endif
+
+#ifdef _USE_ML
+EXTERN __bss_start__:DWORD
+EXTERN __bss_end__:DWORD
+#endif
+
+#ifdef _M_IX86
+.code32
+ASSUME ES:NOTHING, FS:NOTHING, GS:NOTHING
+#else
+.code64
+#endif
+
+PUBLIC FinishRelocation
+
+FinishRelocation:
+    /* Save the registers */
+    push rsi
+    push rdi
+    push rcx
+
+    /* Check if it's loaded with ntldr */
+    cmp byte ptr [BSS_BootedViaNtldr], 0
+    jz .notNtldr
+
+    /* Relocate the freeldr to the NTLDR_TMP_BASE */
```

Comment:
```suggestion
    /* Relocate Freeldr to NTLDR_TMP_BASE */
```

Reply:
Why do we need this two-step relocation, instead of relocating at once from wherever it is down to FREELDR_BASE ?

Reply:
to avoid corruption!